Ltpi retry and mount remote SPI for SAFS

rajaganeshr opened the pull request:
Add changes to mount remote SPI for SAFS
Retry LTPI if training fails on advertise frame alignment

aspeeddylan committed:
cmd: ltpi-v2: Add LTPI register dump

Dump LTPI registers if failed to enter operational state.

v2: Print more information for debugging

Signed-off-by: Dylan Hung <dylan_hung@aspeedtech.com>
Change-Id: I6928ca94190db311fdeb88a9e6d0eb2b13a5ee61

aspeeddylan committed:
cmd: aspeed: ltpi-v2: Enable the configurable advertise timeout

The command argument "-t" is designed for configuring the advertise
timeout in micro-second but was not hooked to the hardware registers.
This patch uses the existing variable "ad_timeout" which is parsed from
the command argument "-t" and configures the LTPI hardware.

Signed-off-by: Dylan Hung <dylan_hung@aspeedtech.com>
Change-Id: I330683067b41ea2a12568d60034dd0e2d2787d1a

rajaganeshr committed:
board: Add retries during LTPI training and mount SPI

Add LTPI advertise timeout as per LTPI v1.1
Add retries to move back to Link Detect frame aligment state on frame errors
Remove LTPI command from boot env
Configure GPIOs to mount remote BIOS SPI

Signed-off-by: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>

ojayanth commented:
Do git revert 
- configs: initialize ltpi before boot 
- Update evb_ast2700.c
- board:a1 Add eSPI GPIO configuration
Commits instead manually deleting changes.

ojayanth commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -50,6 +50,11 @@
 
+/* LTPI */
+#define LTPI_TRAIN_RETRY	(5)
+#define ADVRT_TIMEOUT_US_1_1	"100000"// 100ms LTPI spec 1.1
```
Nit: looks cleaner update in spec version order 1.0 and 1.1

ojayanth commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -50,6 +50,11 @@
 #define ASPEED_SYS_SCRATCH_7FC 0x12C027FC
 #define SYS_SRST		BIT(0)
 
+/* LTPI */
```
Commit message:
- fix typos.
- Update commit message with current scope and use git revert to remove older workarrounds.
-  Please provide some background on the background on this custom ltpi_timeout function.
- if remote BIOS SPI chnages is not linked with timeout please split to two commits.

rajaganeshr replied:
This is WIP, pushing so that I dont have to rely on folders in build machine.

will keep track of these comments and sent one later after

ojayanth commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -325,6 +330,36 @@ int set_board_info(const u8* scm_eeprom_buf, const u8* hpm_eeprom_buf)
+void mount_safs_spi_mux()
+{
+	printf("configuring gpios for eDAF ...\n");
+	/* mount P0 BIOS SPI */
```
Replace P0 with Remote BIOS?

ojayanth commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -325,6 +330,36 @@ int set_board_info(const u8* scm_eeprom_buf, const u8* hpm_eeprom_buf)
+void train_ltpi(int retry)
+{
+        // set espi GPIO strap to low and train ltpi
+        run_command("mw 14c02404 0", 0);
```
int run_command(const char *cmd, int flag); - please handle return , will help for debug incase of failure.

ojayanth commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -325,6 +330,36 @@ int set_board_info(const u8* scm_eeprom_buf, const u8* hpm_eeprom_buf)
+{
+        // set espi GPIO strap to low and train ltpi
+        run_command("mw 14c02404 0", 0);
+        run_command("gpio clear 10", 0);
```
int run_command(const char *cmd, int flag); - please handle return , will help for debug incase of failure.

ojayanth commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -50,6 +50,11 @@
+/* LTPI */
+#define LTPI_TRAIN_RETRY	(5)
+#define ADVRT_TIMEOUT_US_1_1	"100000"// 100ms LTPI spec 1.1
+#define ADVRT_TIMEOUT_US_1_0	"1000"	// 1ms LTPI spec 1.0
```
Looks like this is not used?

ojayanth commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -325,6 +330,36 @@ int set_board_info(const u8* scm_eeprom_buf, const u8* hpm_eeprom_buf)
+{
+	printf("configuring gpios for eDAF ...\n");
+	/* mount P0 BIOS SPI */
+	run_command("gpio set 169", 0); //GPIO V1
```
int run_command(const char *cmd, int flag); - please handle return , will help for debug incase of failure.

ojayanth commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -325,6 +330,36 @@ int set_board_info(const u8* scm_eeprom_buf, const u8* hpm_eeprom_buf)
+	printf("configuring gpios for eDAF ...\n");
+	/* mount P0 BIOS SPI */
+	run_command("gpio set 169", 0); //GPIO V1
+	run_command("gpio set 170", 0); //GPIO V2
```
int run_command(const char *cmd, int flag); - please handle return , will help for debug incase of failure.

ojayanth commented on board/aspeed/evb_ast2700/evb_ast2700.c:
```diff
@@ -325,6 +330,36 @@ int set_board_info(const u8* scm_eeprom_buf, const u8* hpm_eeprom_buf)
+        // set espi GPIO strap to low and train ltpi
+        run_command("mw 14c02404 0", 0);
+        run_command("gpio clear 10", 0);
+        if(run_command("ltpi -t " ADVRT_TIMEOUT_US_1_1, 0) != 0)
```
remove the initial if statement and letting the for loop handle the run_command execution.

rajaganeshr committed:
cmd: ltpi: increase advertise timeout

increase LTPI timeout for retries between link advertisements
fix return code to success if link already operational

Signed-off-by: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>

rajaganeshr committed:
board: add ltpi with 100ms advertise

add ltpi with 100ms advertise
add HCC type check
train only if HPM type is valid for detected SCM

Signed-off-by: Rajaganesh Rathinasabapathi <Rajaganesh.Rathinasabapathi@amd.com>

rajaganeshr commented:
Addressed review comments in pull request #21 
closing this request.
